spec/bolt/transport/docker_spec.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'bolt_spec/conn'
+require 'bolt/transport/docker'
+require 'bolt/target'
+
+require_relative 'shared_examples'
+
+describe Bolt::Transport::Docker do
+  include BoltSpec::Conn
+  let(:hostname) { conn_info('docker')[:host] }
+  let(:docker) { Bolt::Transport::Docker.new }
+  let(:transport_conf) { {} }
+  let(:target) { Bolt::Target.new(hostname, transport_conf) }
+
+  context 'with docker', docker: true do
+    let(:runner) { docker }
+    let(:os_context) { posix_context }
+
+    include_examples 'transport api'
+
+    context 'file errors' do
+      before(:each) do
+        allow_any_instance_of(Bolt::Transport::Docker::Connection).to receive(:write_remote_file).and_raise(
+          Bolt::Node::FileError.new("no write", "WRITE_ERROR")
+        )
+        allow_any_instance_of(Bolt::Transport::Docker::Connection).to receive(:make_tempdir).and_raise(
+          Bolt::Node::FileError.new("no tmpdir", "TEMDIR_ERROR")
+        )
+      end
+
+      include_examples 'transport failures'
+    end
+  end
+
+  context 'with_connection', docker: true do
+    it "fails with an unknown host" do
+      # Test fails differently on Windows due to issues in the docker-api gem.
+      expect {
+        docker.with_connection(Bolt::Target.new('not_a_target')) {}
+      }.to raise_error(Bolt::Node::ConnectError, /Failed to connect to not_a_target: No such container: not_a_target/)
+    end
+  end
+
+  context 'when url is specified', docker: true do
+    let(:transport_conf) { { 'url' => 'tcp://localhost:55555' } }
+
+    it 'uses the url' do
+      expect {
+        docker.with_connection(target) {}
+      }.to raise_error(Bolt::Node::ConnectError, /Connection refused .* 127.0.0.1:55555/)
+    end
+  end
+
+  context 'when options are specified', docker: true do
```

Comment:
can we just set docker true on the describe?

Reply:
Yeah, looks like it.